File: BackendWS/examserver/src/main/java/com/exam/service/QuizService.java
package com.exam.service;

import java.util.List;
import java.util.Set;

import com.exam.entity.exam.Category;
import com.exam.entity.exam.Quiz;

public interface QuizService {
	
	public Quiz addQuiz(Quiz quiz);
	
	public Quiz updateQuiz(Quiz quiz);
	
	public Set<Quiz> getQuizzes();
	
	public Quiz getQuiz(Long quizID);
	
	public void deleteQuiz(Long quizID);

	public List<Quiz> getQuizzesByCategory(Category category);

	public List<Quiz> getActiveQuizzes();
	
	public List<Quiz> getActiveQuizzesByCategory(Category c);
	
	

}

File: FrontendWS/exam-frontend/src/app/services/quiz.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root',
})
export class QuizService {
  constructor(private http: HttpClient) {}

  public quizzes() {
    return this.http.get(`${baseUrl}/quiz/`);
  }

  // add quiz
  public addQuiz(quiz: any) {
    return this.http.post(`${baseUrl}/quiz/`, quiz);
  }

  // delete quiz

  public deleteQuiz(qId: any) {
    return this.http.delete(`${baseUrl}/quiz/${qId}`);
  }

  //get quiz by Id
  public getQuiz(qId: any) {
    return this.http.get(`${baseUrl}/quiz/${qId}`);
  }

  //update quiz
  public updateQuiz(quiz: any) {
    return this.http.put(`${baseUrl}/quiz/`, quiz);
  }

  //get quizzes by category
  public getQuizzesByCategory(cid: any) {
    return this.http.get(`${baseUrl}/quiz/category/${cid}`);
  }

  //get active quizzes
  public getActiveQuizzes() {
    return this.http.get(`${baseUrl}/quiz/active`);
  }

  //get active quizzes by category
  public getActiveQuizzesByCategory(cid: any) {
    return this.http.get(`${baseUrl}/quiz/category/active/${cid}`);
  }
}
